x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/SpatialRelatesUtils.java

```diff
@@ -0,0 +1,112 @@
+        if (crsType == SpatialRelatesFunction.SpatialCrsType.GEO) {
+            var luceneGeometries = LuceneGeometriesUtils.toLatLonGeometry(geometry, true, t -> {});
+            return LatLonGeometry.create(luceneGeometries);
+        } else {
+            var luceneGeometries = LuceneGeometriesUtils.toXYGeometry(geometry, t -> {});
+            return XYGeometry.create(luceneGeometries);
+        }
+    }
+
+    /**
+     * This function is used to convert a spatial constant to a doc-values byte array.
+     * When both left and right sides are constants, we convert the left to a doc-values byte array and the right to a Component2D.
+     */
+    static GeometryDocValueReader asGeometryDocValueReader(SpatialRelatesFunction.SpatialCrsType crsType, Expression expression)
+        throws IOException {
+        Geometry geometry = makeGeometryFromLiteral(expression);
+        if (crsType == SpatialRelatesFunction.SpatialCrsType.GEO) {
+            return asGeometryDocValueReader(
+                CoordinateEncoder.GEO,
+                new GeoShapeIndexer(Orientation.CCW, "SpatialRelatesFunction"),
+                geometry
+            );
+        } else {
+            return asGeometryDocValueReader(CoordinateEncoder.CARTESIAN, new CartesianShapeIndexer("SpatialRelatesFunction"), geometry);
+        }
+
+    }
+
+    /**
+     * Converting shapes into doc-values byte arrays is needed under two situations:
+     * - If both left and right are constants, we convert the right to Component2D and the left to doc-values for comparison
+     * - If the right is a constant and no lucene push-down was possible, we get WKB in the left and convert it to doc-values for comparison
+     */
+    static GeometryDocValueReader asGeometryDocValueReader(CoordinateEncoder encoder, ShapeIndexer shapeIndexer, Geometry geometry)
+        throws IOException {
+        GeometryDocValueReader reader = new GeometryDocValueReader();
+        CentroidCalculator centroidCalculator = new CentroidCalculator();
+        if (geometry instanceof Circle) {
+            // Both the centroid calculator and the shape indexer do not support circles
+            throw new IllegalArgumentException(ShapeType.CIRCLE + " geometry is not supported");
+        }
+        centroidCalculator.add(geometry);
+        reader.reset(GeometryDocValueWriter.write(shapeIndexer.indexShape(geometry), encoder, centroidCalculator));
+        return reader;
+    }
+
+    /**
+     * This function is used in two places, when evaluating a spatial constant in the SpatialRelatesFunction, as well as when
+     * we do lucene-pushdown of spatial functions.
+     */
+    public static Geometry makeGeometryFromLiteral(Expression expr) {
+        Object value = valueOf(expr);
+
+        if (value instanceof BytesRef bytesRef) {
+            if (EsqlDataTypes.isSpatial(expr.dataType())) {
+                return SpatialCoordinateTypes.UNSPECIFIED.wkbToGeometry(bytesRef);
+            } else {
```

Comment:
I think the string parsing is not needed anymore (and those branches are not being exercise by test as expected.). Could be remove those branches from here to simplify it?

Reply:
Yes, we don't need those anymore. I'm removing them now.